server/lib/flow-event.js

```diff
@@ -129,19 +129,20 @@ module.exports = (req, metrics, requestReceivedTime) => {
   const navigationTiming = metrics.navigationTiming;
   if (emitPerformanceEvents && navigationTiming) {
     PERFORMANCE_TIMINGS.forEach(item => {
-      const time = item.timings.reduce((sum, timing) => {
+      const relativeTime = item.timings.reduce((sum, timing) => {
         const from = navigationTiming[timing.from];
         const until = navigationTiming[timing.until];
         if (from >= 0 && until > from) {
           sum += until - from;
         }
         return sum;
       }, 0);
+      const absoluteTime = metrics.flowBeginTime + relativeTime;
 
-      if (time > 0) {
+      if (relativeTime > 0 && isValidTime(absoluteTime, requestReceivedTime)) {
         logFlowEvent({
-          flowTime: time,
-          time: metrics.flowBeginTime + time,
+          flowTime: relativeTime,
+          time: absoluteTime,
           type: `flow.performance.${item.event}`
         }, metrics, req);
       }
```

Comment:
Hmm, maybe. I've decided against it before on the basis that it's not actionable and it doesn't include metrics that were rejected by an earlier validation step (so the data is imprecise). Also, sometimes we get waves of bad data so big that I don't think we want *want* to log it.

As a case in point, for the last day that I had to manually clean our flow data, 7th March, I removed around 400,000 events. Granted, that was a bigger number than usual, and I'm sure that many of them will have been false positives, but it's difficult to be precise when you're dealing with that much data (and the alternative is no data at all because the import fails).

So that, combined with the fact that we wouldn't be doing the same logging for all the other stuff that gets dropped on the floor earlier on, is what makes me question whether it's a good idea.

I'm happy to be persuaded differently though, what do you think?